src/Illuminate/Database/Eloquent/Relations/Concerns/SupportsDefaultModels.php

```diff
@@ -51,7 +51,7 @@ protected function getDefaultFor(Model $parent)
         $instance = $this->newRelatedInstanceFor($parent);
 
         if (is_callable($this->withDefault)) {
-            return call_user_func($this->withDefault, $instance, $parent) ?: $instance;
+            return call_user_func($this->withDefault, $instance);
```

Comment:
Why not passing the parent model anymore?

Reply:
This was unintentional and would explain why tests were failing! Have fixed 😀